lib/services/elasticsearch.js

```diff
@@ -1201,32 +1222,35 @@ class ElasticSearch extends Service {
   _mexecute(esRequest, documents, partialErrors) {
     assertWellFormedRefresh(esRequest);
 
-    if (!this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)) {
-      return Bluebird.reject(new PreconditionError(`Index '${esRequest.index}' and/or collection '${esRequest.type}' don't exist`));
-    }
+    return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)
+      .then(exists => {
+        if (! exists) {
+          throw new PreconditionError(`Index '${esRequest.index}' and/or collection '${esRequest.type}' does not exist`);
+        }
 
-    if (documents.length > this.kuzzle.config.limits.documentsWriteCount) {
-      return Bluebird.reject(new SizeLimitError(`Number of documents exceeds the server configured value (${this.kuzzle.config.limits.documentsWriteCount})`));
-    }
+        if (documents.length > this.kuzzle.config.limits.documentsWriteCount) {
+          return Bluebird.reject(new SizeLimitError(`Number of documents exceeds the server configured value (${this.kuzzle.config.limits.documentsWriteCount})`));
```

Comment:
(nitpicking) since we're now inside a promise chain, we can `throw` instead of returning a rejected promise, for simplicity